### PR TITLE
Add caching of bilinear information

### PR DESCRIPTION
**Pull request by pnuu**
This is a follow-up PR on #300 for moving the cache handling of bilinear resampler from Satpy to Pyresample. The matching Satpy PR is [here](https://github.com/pytroll/satpy/pull/1406). After the Pyresample PR has been merged and a new release is made, the tests on Satpy PR should be run and a new Satpy release should be made.

 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``git diff origin/master **/*py | flake8 --diff`` 
 - [x] Fully documented

### CODE REVIEW

**Review by mraspaud — APPROVED** [review]
LGTM

**Comment by pnuu** [comment]
There was a new PyPy version in Azure that caused the builds to fail. I now disabled all builds for PyPy. The failure happened when building EWA resampler C++ code.

File: pyresample/test/test_bilinear.py [hunk 1]
```diff
@@ -1013,10 +1043,64 @@ def test_lonlat2xyz(self):
...
+        # There are 12 pixels with value 1, all others are NaN
+        res = res.compute()
+        self.assertEqual(np.nansum(res), 12)
+        self.assertEqual(np.isnan(res).sum(), 4)
+
+        # Single array with fill value
+        res = resampler.resample(self.data1, fill_value=0)
+        res = res.compute()
+        self.assertEqual(np.sum(res), 12)
+        self.assertEqual((res == 0).sum(), 4)
```
Comment (mraspaud) [inline pyresample/test/test_bilinear.py:1068]: This could be split into 2 tests

File: pyresample/test/test_bilinear.py [hunk 2]
```diff
@@ -347,11 +347,41 @@ def test_resample_bilinear(self):
...
+        # There should be 12 valid pixels
+        self.assertEqual(self.target_def.size - res.mask.sum(), 12)
+
+        # Two stacked arrays, multiprocessing
+        data = np.dstack((self.data1, self.data2))
+        res = resampler.resample(data, nprocs=2)
+        shp = res.shape
+        self.assertEqual(shp[0:2], self.target_def.shape)
+        self.assertEqual(shp[-1], 2)
+
```
Comment (mraspaud) [inline pyresample/test/test_bilinear.py:379]: This could be split into 3 tests

**Review by mraspaud — COMMENTED** [review]
LGTM. It's always difficult with module refactorings to see what has changed, but since the tests didn't change, I'm confident the functionality stayed the same.

I just have a couple of style comments.

File: .travis.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@ env:
   - NUMPY_VERSION=stable
   - MAIN_CMD='python setup.py'
   - CONDA_DEPENDENCIES='xarray dask toolz Cython pykdtree sphinx cartopy rasterio pillow matplotlib
-    pyyaml pyproj coveralls configobj coverage codecov pytest pytest-cov shapely'
+    pyyaml pyproj coveralls configobj coverage codecov pytest pytest-cov shapely zarr scipy'
```
Comment (djhoese) [inline .travis.yml:8]: Why is scipy required? I thought it wasn't needed since we moved to pykdtree for all the KDTree stuff?

Comment (pnuu) [reply .travis.yml:8]: The legacy Numpy based resampler uses Scipy to query in parallel.

**Comment by pnuu** [comment]
This PR works with both the current `master` and latest release of Satpy, so merging and (later) releasing won't break Satpy. Merging.

**Pull request merged**